pkgs/development/python-modules/nbxmpp/default.nix

```diff
@@ -1,26 +1,31 @@
-{ lib, buildPythonPackage, fetchzip, gobject-introspection, idna, libsoup, precis-i18n, pygobject3, pyopenssl }:
+{ lib, buildPythonPackage, pythonOlder, fetchFromGitLab
+, gobject-introspection, idna, libsoup, precis-i18n, pygobject3, pyopenssl
+}:
 
-let
+buildPythonPackage rec {
   pname = "nbxmpp";
-  version = "1.0.2";
-  name = "${pname}-${version}";
-in buildPythonPackage {
-  inherit pname version;
+  version = "2.0.2";
+
+  disabled = pythonOlder "3.7";
+
   # Tests aren't included in PyPI tarball.
-  src = fetchzip {
-    name = "${name}.tar.bz2";
-    url = "https://dev.gajim.org/gajim/python-nbxmpp/repository/archive.tar.bz2?"
-        + "ref=${name}";
-    sha256 = "1rhzsakqrybzq5j5b9400wjd14pncph47c1ggn5a6f3di03lk4az";
+  src = fetchFromGitLab {
+    domain = "dev.gajim.org";
+    owner = "gajim";
+    repo = "python-nbxmpp";
+    rev = "nbxmpp-${version}";
```

Comment:
```suggestion
    rev = "${pname}-${version}";
```

Reply:
Thanks a lot for your time having a look at this pull request!

However, due to this suggested change: I don't want to come off as being ungracious, but is this really necessary? This neither increases readability nor avoids redundancies when updating, since the name is unlikely to change.